src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -0,0 +1,128 @@
+from typing import TYPE_CHECKING, Union
+
+import numpy as np
+
+from ..utils import logging
+from .base import Pipeline
+
+
+if TYPE_CHECKING:
+    from ..models.auto import AutoFeatureExtractor
+
+logger = logging.get_logger(__name__)
+
+
+def ffmpeg_read(bpayload: bytes, sampling_rate: int) -> np.array:
+    """
+    Helper function to read an audio file through ffmpeg.
+    """
+    ar = f"{sampling_rate}"
+    ac = "1"
+    format_for_conversion = "f32le"
+    ffmpeg_command = [
+        "ffmpeg",
+        "-i",
+        "pipe:0",
+        "-ac",
+        ac,
+        "-ar",
+        ar,
+        "-f",
+        format_for_conversion,
+        "-hide_banner",
+        "-loglevel",
+        "quiet",
+        "pipe:1",
+    ]
+
+    ffmpeg_process = subprocess.Popen(ffmpeg_command, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+    output_stream = ffmpeg_process.communicate(bpayload)
+    out_bytes = output_stream[0]
+
+    audio = np.frombuffer(out_bytes, np.float32)
+    if audio.shape[0] == 0:
+        raise ValueError("Malformed soundfile")
+    return audio
+
+
+class AutomaticSpeechRecognitionPipeline(Pipeline):
+    """
+    Pipeline that aims at extracting spoken text contained within some audio.
+
+    The input can be either a raw waveform or a audio file. In case of the audio file, ffmpeg should
+    be installed for the read function requires it (to support many formats).
+    """
+
+    def __init__(self, feature_extractor: "AutoFeatureExtractor", *args, **kwargs):
+        """
+        Arguments:
+            model (:obj:`~transformers.PreTrainedModel` or :obj:`~transformers.TFPreTrainedModel`):
+                The model that will be used by the pipeline to make predictions. This needs to be a model inheriting from
+                :class:`~transformers.PreTrainedModel` for PyTorch and :class:`~transformers.TFPreTrainedModel` for
+                TensorFlow.
+            tokenizer (:obj:`~transformers.PreTrainedTokenizer`):
+                The tokenizer that will be used by the pipeline to encode data for the model. This object inherits from
+                :class:`~transformers.PreTrainedTokenizer`.
+            feature_extractor (:obj:`~transformers.AutoFeatureExtractor`):
+                The feature extractor that will be used by the pipeline to encode waveform for the model.
+            modelcard (:obj:`str` or :class:`~transformers.ModelCard`, `optional`):
+                Model card attributed to the model for this pipeline.
+            framework (:obj:`str`, `optional`):
+                The framework to use, either :obj:`"pt"` for PyTorch or :obj:`"tf"` for TensorFlow. The specified framework
+                must be installed.
+
+                If no framework is specified, will default to the one currently installed. If no framework is specified and
+                both frameworks are installed, will default to the framework of the :obj:`model`, or to PyTorch if no model
+                is provided.
```

Comment:
It should probably just fail if something other than `pt` is passed in, as some very specific PyTorch instructions are set in the call methods, and the models don't have a TensorFlow counterpart.

See an example of this in the `TableQuestionAnsweringPipeline`:
https://github.com/huggingface/transformers/blob/cfd2eaa8cf82da8581825c6592b66d2789c5bc53/src/transformers/pipelines/table_question_answering.py#L86-L87

Reply:
Good point !